tests/website.py

```diff
@@ -16,6 +16,9 @@ def _pre_setup_10_website(self):
         self.deployment.add("haproxy")
         self.deployment.relate("jenkins:website", "haproxy:reverseproxy")
         self.deployment.expose("haproxy")
+        self.deployment.add("nrpe")
+        self.deployment.relate("jenkins:nrpe-external-master",
+                               "nrpe:nrpe-external-master")
```

Comment:
Yes, the test suite will get longer, but I have a few comments:

1) Tests will be more granular and independent. This means that if you are touching nrpe-related code you most likely want to run only the nrpe relation tests, which will be as fast as it could be. You'll want to run the full suite just at the end, and chances that you broke something unrelated should be low (and even it that case, if tests are granular it should be fast to run the one single test that failed).

2) I think there is a big room for improvement in the integration test tool chain (bundletester/amulet/juju-deployer), which currently does a lot of extra work without re-using what it did before (there are also various bottlenecks that can be removed). I hope that a re-designed tool chain based on the new asyncio juju client library will greatly improve integration test speed (or at least make it faster to re-run them in some situations), so your concern should go away in the medium term.